Could not send a password-protected message if 'to' doesn't contain recipients without pub keys

It relates to the enterprise issue with password-protected messages

```
[SUMMARY]
com.flowcrypt.email.util.exception.ManualHandledException: Handled manually:
com.flowcrypt.email.util.exception.ApiException: API error: code = 400, message = Bad Request


[STACK_TRACE]
com.flowcrypt.email.util.exception.ManualHandledException: Handled manually:
        at com.flowcrypt.email.util.exception.ExceptionUtil$Companion.handleError(ExceptionUtil.kt:30)
        at com.flowcrypt.email.jetpack.workmanager.HandlePasswordProtectedMsgWorker$handleExceptionsForMessage$2.invokeSuspend(HandlePasswordProtectedMsgWorker.kt:231)
        at com.flowcrypt.email.jetpack.workmanager.HandlePasswordProtectedMsgWorker$handleExceptionsForMessage$2.invoke(Unknown Source:8)
        at com.flowcrypt.email.jetpack.workmanager.HandlePasswordProtectedMsgWorker$handleExceptionsForMessage$2.invoke(Unknown Source:4)
        at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:89)
        at kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(Builders.common.kt:165)
        at kotlinx.coroutines.BuildersKt.withContext(Unknown Source:1)
        at com.flowcrypt.email.jetpack.workmanager.HandlePasswordProtectedMsgWorker.handleExceptionsForMessage(HandlePasswordProtectedMsgWorker.kt:230)
        at com.flowcrypt.email.jetpack.workmanager.HandlePasswordProtectedMsgWorker.access$handleExceptionsForMessage(HandlePasswordProtectedMsgWorker.kt:71)
        at com.flowcrypt.email.jetpack.workmanager.HandlePasswordProtectedMsgWorker$prepareAndUploadPasswordProtectedMsgsToFES$2.invokeSuspend(HandlePasswordProtectedMsgWorker.kt:207)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.UndispatchedCoroutine.afterResume(CoroutineContext.kt:147)
        at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.kt:102)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
        at kotlinx.coroutines.UndispatchedCoroutine.afterResume(CoroutineContext.kt:147)
        at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.kt:102)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
        at kotlinx.coroutines.internal.LimitedDispatcher.run(LimitedDispatcher.kt:39)
        at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:95)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
Caused by: com.flowcrypt.email.util.exception.ApiException: API error: code = 400, message = Bad Request
        at com.flowcrypt.email.api.retrofit.base.BaseApiRepository$DefaultImpls.getResult(BaseApiRepository.kt:64)
        at com.flowcrypt.email.api.retrofit.base.BaseApiRepository$getResult$1.invokeSuspend(Unknown Source:18)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        ... 7 more
```

```json
{
   "code":400,
   "message":"Missing recipient",
   "details":"For more details, see FlowCrypt Enterprise Server logs."
}
```

--- a/FlowCrypt/src/main/java/com/flowcrypt/email/extensions/javax/mail/internet/MimeMessageExt.kt
+++ b/FlowCrypt/src/main/java/com/flowcrypt/email/extensions/javax/mail/internet/MimeMessageExt.kt
@@ -5,7 +5,6 @@
 
 package com.flowcrypt.email.extensions.javax.mail.internet
 
-import javax.mail.Address
 import javax.mail.Message
 import javax.mail.internet.InternetAddress
 import javax.mail.internet.MimeMessage
@@ -24,13 +23,4 @@ fun MimeMessage.getAddresses(type: Message.RecipientType): List<String> {
 fun MimeMessage.getFromAddress(): InternetAddress {
   return (from.first() as? InternetAddress)
     ?: throw IllegalStateException("'from' address is undefined")
-}
-
-fun MimeMessage.getMatchingRecipients(
-  type: Message.RecipientType,
-  list: Iterable<String>
-): Array<Address> {
-  return getRecipients(type)?.filter {
-    (it as? InternetAddress)?.address?.lowercase() in list
-  }?.toTypedArray() ?: emptyArray()
 }

--- a/FlowCrypt/src/main/java/com/flowcrypt/email/jetpack/workmanager/HandlePasswordProtectedMsgWorker.kt
+++ b/FlowCrypt/src/main/java/com/flowcrypt/email/jetpack/workmanager/HandlePasswordProtectedMsgWorker.kt
@@ -21,9 +21,7 @@ import com.flowcrypt.email.database.MessageState
 import com.flowcrypt.email.database.entity.AccountEntity
 import com.flowcrypt.email.database.entity.AttachmentEntity
 import com.flowcrypt.email.database.entity.MessageEntity
-import com.flowcrypt.email.extensions.javax.mail.internet.getAddresses
 import com.flowcrypt.email.extensions.javax.mail.internet.getFromAddress
-import com.flowcrypt.email.extensions.javax.mail.internet.getMatchingRecipients
 import com.flowcrypt.email.extensions.kotlin.toInputStream
 import com.flowcrypt.email.jetpack.viewmodel.AccountViewModel
 import com.flowcrypt.email.jetpack.workmanager.base.BaseMsgWorker
@@ -128,10 +126,12 @@ class HandlePasswordProtectedMsgWorker(context: Context, params: WorkerParameter
             val plainMimeMsgWithAttachments = getMimeMessage(account, msgEntity, attachments)
 
             //get recipients that will be used to create password-encrypted msg
-            val toCandidates = getRecipients(plainMimeMsgWithAttachments, Message.RecipientType.TO)
-            val ccCandidates = getRecipients(plainMimeMsgWithAttachments, Message.RecipientType.CC)
+            val toCandidates =
+              plainMimeMsgWithAttachments.getRecipients(Message.RecipientType.TO) ?: emptyArray()
+            val ccCandidates =
+              plainMimeMsgWithAttachments.getRecipients(Message.RecipientType.CC) ?: emptyArray()
             val bccCandidates =
-              getRecipients(plainMimeMsgWithAttachments, Message.RecipientType.BCC)
+              plainMimeMsgWithAttachments.getRecipients(Message.RecipientType.BCC) ?: emptyArray()
 
             if (toCandidates.isEmpty() && ccCandidates.isEmpty() && bccCandidates.isEmpty()) {
               throw IllegalStateException("Wrong password-protected implementation")
@@ -143,12 +143,14 @@ class HandlePasswordProtectedMsgWorker(context: Context, params: WorkerParameter
             val idToken = getGoogleIdToken()
             val replyToken = fetchReplyToken(apiRepository, domain, idToken)
             val replyInfoData = ReplyInfoData(
-              sender = fromAddress.address,
+              sender = fromAddress.address.lowercase(),
               recipient = (toCandidates + ccCandidates + bccCandidates)
-                .mapNotNull { (it as? InternetAddress)?.address }
+                .mapNotNull { (it as? InternetAddress)?.address?.lowercase() }
                 .filterNot {
                   it.equals(fromAddress.address, true)
-                },
+                }
+                .toHashSet()
+                .toList(),
               subject = plainMimeMsgWithAttachments.subject,
               token = replyToken
             )
@@ -198,8 +200,7 @@ class HandlePasswordProtectedMsgWorker(context: Context, params: WorkerParameter
 
             updateExistingMimeMsgWithUrl(msgEntity, fesUrl)
           } else {
-            roomDatabase.msgDao()
-              .updateSuspend(msgEntity.copy(state = MessageState.QUEUED.value))
+            roomDatabase.msgDao().updateSuspend(msgEntity.copy(state = MessageState.QUEUED.value))
           }
           MessagesSenderWorker.enqueue(applicationContext)
         } catch (e: Exception) {
@@ -312,19 +313,6 @@ class HandlePasswordProtectedMsgWorker(context: Context, params: WorkerParameter
         decryptWhenForward = true
       )
     }
-
-  private suspend fun getRecipients(
-    mimeMsgWithAttachments: MimeMessage,
-    type: Message.RecipientType
-  ) = withContext(Dispatchers.IO) {
-    mimeMsgWithAttachments.getMatchingRecipients(
-      type = type,
-      list = GeneralUtil.getRecipientsWithoutUsablePubKeys(
-        context = applicationContext,
-        emails = mimeMsgWithAttachments.getAddresses(type)
-      )
-    )
-  }
 
   private suspend fun updateExistingMimeMsgWithUrl(
     msgEntity: MessageEntity,

